homeassistant/components/knx/schema.py

```diff
@@ -24,6 +25,32 @@
     ColorTempModes,
 )
 
+##################
+# KNX VALIDATORS
+##################
+
+ga_validator = vol.Any(
+    cv.matches_regex(GroupAddress.ADDRESS_RE),
+    vol.All(vol.Coerce(int), vol.Range(min=1, max=65535)),
```

Comment:
This looks of. If the regexp above happen to match a single int. That won't be coerced to int. It will stay a string . The first matched path of a vol.Any will be taken.

Reply:
This is ok. `GroupAddress.__init__` expects `Optional[Union["GroupAddress", str, int, Tuple[int, int]]]`
str is used for 2- and 3-level numbering Schema. That's the most common ones with the "/" separator. 
Integers are for the "free-Level" numbering Schema. A string would work too ("123") but this way we can check it's range. 
(int Tuple is for raw bytes - this is used internally.)

Reply:
So then drop the coerce. If anything coerce it to string first?

Reply:
What i meant is, if that regexp accepts a single int (in string form) it's pointless to accept int and the coercing will never occure.

Reply:
Got it! Removed coerce. 
Thank you for pointing this out! 👍

Reply:
@elupus The regex cannot match a single int or string. It will only allow the 2 or 3 level "/" separated formats. Everything else needs to be handled by the range check and for that the coerce I think makes sense, doesn't it?

Reply:
Yes if that is the case. An alternative would be to let validation construct a group object, catching any value error exceptions. 

Then moving parsing into that object. Less places to handle parsing.

But I'm okey with it either way.